server/webhook.go

```diff
@@ -57,15 +58,63 @@ func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
 	}
 
 	switch webhook.Event {
+	case zoom.EventTypeMeetingStarted:
+		p.handleMeetingStarted(w, r, b)
 	case zoom.EventTypeMeetingEnded:
 		p.handleMeetingEnded(w, r, b)
 	case zoom.EventTypeValidateWebhook:
 		p.handleValidateZoomWebhook(w, r, b)
+	case zoom.EventTypeRecordingCompleted:
+		p.handleRecordingCompleted(w, r, b)
+	case zoom.EventTypeTranscriptCompleted:
+		p.handleTranscriptCompleted(w, r, b)
 	default:
 		w.WriteHeader(http.StatusOK)
 	}
 }
 
+func (p *Plugin) handleMeetingStarted(w http.ResponseWriter, r *http.Request, body []byte) {
+	var webhook zoom.MeetingWebhook
+	if err := json.Unmarshal(body, &webhook); err != nil {
+		p.API.LogError("Error unmarshaling meeting webhook", "err", err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	meetingID, err := strconv.Atoi(webhook.Payload.Object.ID)
+	if err != nil {
+		p.API.LogError("Failed to get meeting ID", "err", err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	channelID, appErr := p.fetchChannelForMeeting(meetingID)
+	if appErr != nil {
+		http.Error(w, appErr.Error(), appErr.StatusCode)
+		return
+	}
+
+	if channelID == "" {
+		return
+	}
+
+	botUser, appErr := p.API.GetUser(p.botUserID)
+	if appErr != nil {
+		p.API.LogError("Failed to get bot user", "err", appErr.Error())
+		http.Error(w, appErr.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if postMeetingErr := p.postMeeting(botUser, meetingID, webhook.Payload.Object.UUID, channelID, "", webhook.Payload.Object.ID); postMeetingErr != nil {
+		p.API.LogError("Failed to post the zoom message in the channel", "err", postMeetingErr.Error())
+		http.Error(w, postMeetingErr.Error(), http.StatusBadRequest)
+		return
+	}
+
+	p.trackMeetingStart(p.botUserID, telemetryStartSourceCommand)
+	p.trackMeetingType(p.botUserID, false)
```

Comment:
We're not really "starting" a meeting here. And we'll get this webhook message when we start through a slash command anyway, which would then have duplicate telemetry events

I think maybe we just avoid creating telemetry events in this case. Maybe we want to capture usage of the recording and transcription features though

Reply:
This is generating the telemetry because these are meetings posts created when a meeting started from outside mattermost and a channel is subscribe to that meeting.

Reply:
Done